lib/beautiful/gtk.lua

```diff
@@ -0,0 +1,178 @@
+---------------------------------------------------------------------------
+--- Querying current GTK+ 3 theme via GtkStyleContext.
+--
+-- @author Yauheni Kirylau &lt;yawghen@gmail.com&gt;
+-- @copyright 2016-2017 Yauheni Kirylau
+-- @module beautiful.gtk
+---------------------------------------------------------------------------
+local gears_debug = require("gears.debug")
+local unpack = unpack or table.unpack -- luacheck: globals unpack (compatibility with Lua 5.1)
+
+local gtk = {
+  cached_theme_variables = nil
+}
+
+
+local function convert_gtk_color_to_hex(gtk_color)
+  local rgb = gtk_color:to_string()
+  local m = rgb:gmatch("[0-9]+")
+  return m and (
+    "#" .. string.format("%2.2x", m()) .. string.format("%2.2x", m()) .. string.format("%2.2x", m())
+  )
+end
+
+
+local function lookup_gtk_color_to_hex(_style_context, color_name)
+  local color = _style_context:lookup_color(color_name)
+  if not color then
+      return nil
+  end
+  local result = convert_gtk_color_to_hex(color)
+  --color:free()
+  return result
+end
+
+local function get_gtk_color_property_to_hex(_style_context, property_name, state)
+    local property = _style_context:get_property(property_name, state)
```

Comment:
I don't really know this API, but the documentation says:
> Note that passing a state other than the current state of context is not recommended unless the style context has been saved with gtk_style_context_save().

https://developer.gnome.org/gtk3/stable/GtkStyleContext.html#gtk-style-context-get-property

Does that matter here? Why not?